backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt

```diff
@@ -0,0 +1,211 @@
+import net.skyscanner.backpack.compose.divider.BpkDivider
+import net.skyscanner.backpack.compose.segmentedcontrol.BpkSegmentedControlStyle
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
+import net.skyscanner.backpack.compose.tokens.BpkBorderSize
+import net.skyscanner.backpack.compose.tokens.BpkElevation
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.internal.BpkSegmentedControlColors
+
+private const val BpkAnimationDurationMs = 50
+private val BpkSegmentedControlHeight = 32.dp
+private val ButtonShape = RoundedCornerShape(BpkBorderRadius.Sm)
+
+@Composable
+internal fun BpkSegmentedControlImpl(
+    buttonContents: List<String>,
+    onItemClick: (Int) -> Unit,
+    selectedIndex: Int,
+    shadow: Boolean,
+    type: BpkSegmentedControlStyle,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = modifier
+            .applyShadow(shadow)
+            .clip(ButtonShape)
+            .height(BpkSegmentedControlHeight)
+            .fillMaxWidth()
+            .background(type.toColor())
+            .selectableGroup(),
+    ) {
+        buttonContents.forEachIndexed { index, content ->
+            val isSelected = selectedIndex == index
+            BpkSegmentedControlButton(
+                modifier = Modifier.weight(1f),
+                isSelected = isSelected,
+                type = type,
+                content = content,
+                onItemClick = { onItemClick(index) },
+            )
+            if (shouldShowDivider(index, buttonContents.lastIndex, selectedIndex, isSelected)) {
+                BpkDivider(
+                    modifier = Modifier
+                        .fillMaxHeight()
+                        .width(BpkBorderSize.Sm),
+                )
+            }
+        }
+    }
+}
+
+private fun shouldShowDivider(
+    index: Int,
+    lastIndex: Int,
+    selectedInt: Int,
+    isSelected: Boolean,
+): Boolean {
+    val isLastItem = index == lastIndex
+    val isNextSelected = index + 1 == selectedInt
+    return !isLastItem && !isNextSelected && !isSelected
+}
+
+@Composable
+private fun BpkSegmentedControlButton(
+    isSelected: Boolean,
+    type: BpkSegmentedControlStyle,
+    content: String,
+    onItemClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Box(
+        modifier = modifier
+            .fillMaxSize()
+            .applyBackground(
+                selected = isSelected,
+                background = getButtonColor(type),
+            )
+            .padding(horizontal = BpkSpacing.Md)
+            .selectable(
+                selected = isSelected,
+                role = Role.RadioButton,
+                interactionSource = remember { MutableInteractionSource() },
+                indication = null,
```

Comment:
is it intentional that there's no ripple? we should always have some touch feedback

Reply:
at first I added it, then I removed it while playing with shadow, good spot